Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -594,8 +594,10 @@ void NetworkProcessProxy::terminateWebProcess(WebCore::ProcessIdentifier webProc
 
 void NetworkProcessProxy::processHasUnresponseServiceWorker(WebCore::ProcessIdentifier processIdentifier)
 {
-    if (auto process = WebProcessProxy::processForIdentifier(processIdentifier))
+    if (auto process = WebProcessProxy::processForIdentifier(processIdentifier)) {
+        process->endServiceWorkerBackgroundProcessing();
```

Comment:
I _think_ from reading the code this is necessary, although maybe I should fold in dropping the background processing activity in to `disableRemoteWorkers`?